### PR TITLE
Atom feed

### PR DESCRIPTION
**Pull request by efexen**
Hey Andy,

This should resolve issue #50 

It adds a new atom feed for the videos page and includes an auto discovery link in the layout so people can just paste the url http://rubyvideos.com and the feed readers should pick up the feed url correctly. I've also added a link to the feed in the footer next to the social icons to advertise the fact that a feed is available.

I wasn't 100% sure on the content so decided to add the embedded video and full description so as to give most value directly in the feed, you ok with this or would you rather it was a teaser and the visitors would come to the site to view the video?

I've tested this in Feedly and seemed to render very nicely.

The experimental style of the project and some of the choices like cells gem and slim are new to me so have a look and see if you think I've managed to keep to the style of the project.

Let me know what you think,
.FxN

### CODE REVIEW

**Comment by andypike** [comment]
Thanks for the PR, it looks great, just a couple of things if that's ok:

A couple of style things, I prefer double quotes over single quotes and also hash rockets over 1.9 hash style, it would be awesome if you could update those please.

Also, would you mind adding a request spec for `/videos.atom`? So, given some known videos, check that the response is in a correctly formed atom format.

That would be amazing, thanks you vey much!

Andy :heart:

**Comment by efexen** [comment]
No problem I'll fix those bits and have a look at adding a request spec.

I didn't add a spec initially as I couldn't find existing specs to test rendering so wasn't sure on strategy

**Comment by andypike** [comment]
Sure thing. So at the moment there are just feature specs that drive the browser and unit specs. I think a request spec is best for this as there is no real UI requirement so just create a `requests` folder under `specs` and add a normal rails style request spec. I'm happy to help out if you need any assistance so just let me know :smile: 

Thanks again for the PR!

**Comment by efexen** [comment]
Added 2 simple request tests that validate the feed and an entry for the required parts as detailed here: http://atomenabled.org/developers/syndication/

Tidied some of the styling as requested

Let me know if the specs are appropriate

File: app/views/videos/index.atom.slim [hunk 1]
```diff
@@ -0,0 +1,15 @@
+doctype xml
+feed xmlns="http://www.w3.org/2005/Atom"
+
+  title Ruby Videos
+  id = root_url
```
Comment (andypike) [inline app/views/videos/index.atom.slim]: Can you please remove the space between `id` and `=`?

File: spec/requests/videos_feed_spec.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
+require "rails_helper"
+
+RSpec.describe "Videos Feed", :type => :request do
```
Comment (andypike) [inline spec/requests/videos_feed_spec.rb]: Can you remove the `:type` tag as we use `infer_spec_type_from_file_location!` so this isn't required. Thx.

File: spec/requests/videos_feed_spec.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+  let!(:video) do
+    create(:published_video,
+           :updated_at => DateTime.new(2015, 6, 9))
```
Comment (andypike) [inline spec/requests/videos_feed_spec.rb]: A slight format change here please to the following:

```
create(:published_video,
  :updated_at => DateTime.new(2015, 6, 9)
)
```

File: spec/requests/videos_feed_spec.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  end
+
+  before do
+    get videos_path, format: :atom
```
Comment (andypike) [inline spec/requests/videos_feed_spec.rb]: Can you use a hash rocket here please?

**Comment by andypike** [comment]
Thanks for the update buddy. I've added a few comments (sorry for being so pedantic). :heart:

File: app/views/videos/index.atom.slim (hunk 1 above)
Comment (efexen) [inline app/views/videos/index.atom.slim]: Sorry must have missed this when tidying up

File: spec/requests/videos_feed_spec.rb (hunk 2 above)
Comment (efexen) [inline spec/requests/videos_feed_spec.rb]: :+1:

File: spec/requests/videos_feed_spec.rb (hunk 3 above)
Comment (efexen) [inline spec/requests/videos_feed_spec.rb]: :+1:

File: spec/requests/videos_feed_spec.rb (hunk 4 above)
Comment (efexen) [inline spec/requests/videos_feed_spec.rb]: Good spot! :+1:

**Comment by efexen** [comment]
Latest commit should address the styling issues.

Not to worry, better to keep it consistent and clean. Hunting for missing hash rockets turned out to be quite hard, have a lot more experience doing it other way around :wink:

**Comment by andypike** [comment]
Thanks man, all merged :smile: I use continuous deployment so once the build is finished it will be deployed to production. :shipit: 

Thanks again!!

**Comment by efexen** [comment]
Awesome can add it to my Feedly in the morning then :smile:

**Pull request merged**